src/java/org/apache/cassandra/db/guardrails/GuardrailsMBean.java

```diff
@@ -651,4 +651,56 @@
      * @param value {@code true} if 0 default TTL on TWCS tables is allowed, {@code false} otherwise.
      */
     void setZeroTTLOnTWCSEnabled(boolean value);
+
+    /**
+     * @param currentTimestamp the current timestamp
+     * @return The highest acceptable timestamp before triggering a warning
+     */
+    long getMaximumTimestampWarnThreshold(long currentTimestamp);
+
+    /**
+     * Sets the warning upper bound for user supplied timestamps
+     *
+     * @param durationSeconds currentTimestamp + TimeUnit.SECONDS.toMicros(duration)
+     */
+    void setMaximumTimestampWarnThreshold(int durationSeconds);
+
+    /**
+     * @param currentTimestamp the current timestamp
+     * @return The highest acceptable timestamp before triggering a failure
+     */
+    long getMaximumTimestampFailThreshold(long currentTimestamp);
```

Comment:
Probably it would make sense to have JMX getters returning the value of the property as it appears in config, instead of the ones relative to the current timestamp. Those getters would be the counterpart of the setters, which seem to accept the yaml property as argument.

Reply:
Perhaps we can try to homogenize the JMX methods and config so they only deal with the config properties as they are expressed in yaml, and encapsulate in `Guardrails.java` the part about making those thresholds relative to the current time. More or less [this way](https://github.com/adelapena/cassandra/commit/514be38817e7e21e3da9244fcac338c1b512fbee). wdyt?

Reply:
I think the challenge here is it will fail `JMXStandardsTest`

Reply:
Ah I think I see in your patch you may have addressed that. Will take a closer look.